### DISCUSSION TITLE
'RefCallBack' is not assignable to parameter of type 'RefObject<HTMLElement>'

### DISCUSSION THREAD

**Discussion by onlyhasbi** [react-hook-form/react-hook-form#10181] [Q&A]
register component
`const { register, setFocus, setValue } = useForm();
const { ref, onChange, onBlur, name } = register("title");`

i put ref here
`useClickAway(ref, isEditTitle, handleDoneEdit);`

and here
`<input
              ref={ref}
              name={name}
              onChange={onChange}
              onBlur={onBlur}
              type="text"
              disabled={!isEditTitle}
              className={clsx([
                "relative w-7/12 ml-[36.33px] mr-[23px] text-4xl font-bold text-generalblack",
                "bg-transparent border-none outline-none ring-0  focus:ring-0 p-0",
                { "underline underline-offset-[.9rem]": isEditTitle },
              ])}
            />`

and error message showing 'RefCallBack' is not assignable to parameter of type 'RefObject<HTMLElement>'
anyone have the same problem ?

**Comment by onlyhasbi** [comment]
the error gone when i try to make 2 ref and put both in the input like this,

const { ref: refRegister, onChange, onBlur, name } = register("title");
const refTitle = useRef(null);

`<input
              ref={(ref) => {
                refTitle;
                refRegister(ref);
              }}
              name={name}
              onChange={onChange}
              onBlur={onBlur}
              type="text"
              disabled={!isEditTitle}
              className={clsx([
                "relative w-7/12 ml-[36.33px] mr-[23px] text-4xl font-bold text-generalblack",
                "bg-transparent border-none outline-none ring-0  focus:ring-0 p-0",
                { "underline underline-offset-[.9rem]": isEditTitle },
              ])}
            />`
            
         i don't know its correct or not, but then the error gone

**Comment by lamhungypl** [comment]
please provide a CSB
